packages/connection-model/lib/model.js

```diff
@@ -805,11 +817,42 @@ Connection = AmpersandModel.extend({
     }
 
     if (attrs.authStrategy === 'KERBEROS') {
-      if (!attrs.kerberosServiceName) {
-        attrs.kerberosServiceName = KERBEROS_SERVICE_NAME_DEFAULT;
-      }
+      const authProperties = attrs.authMechanismProperties || {};
+      this.kerberosServiceName = defaultTo(
+        attrs.kerberosServiceName,
+        defaultTo(attrs.gssapiServiceName, authProperties.SERVICE_NAME)
+      );
+
+      this.kerberosServiceRealm = defaultTo(
+        attrs.kerberosServiceRealm,
+        defaultTo(attrs.gssapiServiceRealm, authProperties.SERVICE_REALM)
+      );
 
-      this.kerberosServiceName = attrs.kerberosServiceName;
+      this.kerberosCanonicalizeHostname = defaultTo(
+        attrs.kerberosCanonicalizeHostname,
+        defaultTo(
+          attrs.gssapiCanonicalizeHostName,
+          defaultTo(
+            attrs.gssapicanonicalizehostname,
+            defaultTo(authProperties.CANONICALIZE_HOST_NAME, authProperties.gssapiCanonicalizeHostName)
+          )
+        )
+      );
+
+      delete attrs.gssapiServiceName;
+      this.gssapiServiceName = undefined;
+      delete attrs.gssapiServiceRealm;
+      this.gssapiServiceRealm = undefined;
+      delete attrs.gssapiCanonicalizeHostName;
+      this.gssapiCanonicalizeHostName = undefined;
+      delete attrs.gssapicanonicalizehostname;
+
+      if (attrs.authMechanismProperties) {
+        delete attrs.authMechanismProperties.SERVICE_NAME;
+        delete attrs.authMechanismProperties.SERVICE_REALM;
+        delete attrs.authMechanismProperties.CANONICALIZE_HOST_NAME;
+        delete attrs.authMechanismProperties.gssapiCanonicalizeHostName;
+      }
```

Comment:
I know it sounds silly in a file where the shortest function is 1000 lines of code .. but could we extract this part in a function :)?

Reply:
Did it!